### PR TITLE
Add: automate OCP-23501 for new upgrade scenario

### PR DESCRIPTION
**Pull request by duanwei33**
[Pass log](https://privatebin-it-iso.int.open.paas.redhat.com/?cb0d207136f29daf#8L/q1UfV1cuNeR210LdYcZlcoI2CKMnlToUEMf6uAAc=)

### CODE REVIEW

**Comment by duanwei33** [comment]
@qinpingli @chao007 
Please also take a look at this upgrade case.

File: testdata/storage/misc/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    matchLabels:
+      app: hello-openshift
+  template:
+    metadata:
+      labels:
+        app: hello-openshift
+    spec:
+      containers:
+      - name: hello-openshift
+        image: aosqe/hello-openshift
```
Comment (qinpingli) [inline testdata/storage/misc/deployment.yaml:17]: How about use the image aosqe/storage+digest?

File: features/upgrade/storage/upgrade.feature [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    Given the expression should be true> cluster_operator('storage').condition(type: 'Available')['status'] == "True"
+    Given the expression should be true> cluster_operator('storage').condition(type: 'Degraded')['status'] == "False"
+    Given the expression should be true> cluster_operator('storage').condition(type: 'Upgradeable')['status'] == "True"
+
+    # There should be one and only one default storage class
+    When I run the :get client command with:
+      | resource | storageclass |
+    Then the step should succeed
+    And the output should contain:
+      | (default) |
```
Comment (qinpingli) [inline features/upgrade/storage/upgrade.feature:20]: I prefer to check there is on default sc here.

File: features/upgrade/storage/upgrade.feature [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    When I run the :get client command with:
+      | resource | storageclass |
+      | o        | yaml         |
+    Then the step should succeed
+    And the output should contain 1 times:
+      | is-default-class: "true" |
```
Comment (qinpingli) [inline features/upgrade/storage/upgrade.feature:26]: In here we can check if the default sc is managed by cluster storage operator.

File: features/upgrade/storage/upgrade.feature [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
```
Comment (qinpingli) [inline features/upgrade/storage/upgrade.feature:76]: same with prepare case.

File: features/upgrade/storage/upgrade.feature [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  @upgrade-check
+  @users=upuser1,upuser2
+  @admin
+  Scenario: Cluster operator storage should be in correct status and dynamic provisioning should work well after upgrade - prepare
+    Given I switch to cluster admin pseudo user
+    # Check cluster operator storage should be in correct status
+    Given the expression should be true> cluster_operator('storage').condition(type: 'Progressing')['status'] == "False"
+    Given the expression should be true> cluster_operator('storage').condition(type: 'Available')['status'] == "True"
+    Given the expression should be true> cluster_operator('storage').condition(type: 'Degraded')['status'] == "False"
+    Given the expression should be true> cluster_operator('storage').condition(type: 'Upgradeable')['status'] == "True"
```
Comment (qinpingli) [inline features/upgrade/storage/upgrade.feature:63]: I prefer to add a step to check if the storage cluster operator version is same with the clusterversion.

File: features/upgrade/storage/upgrade.feature (hunk 2 above)
Comment (qinpingli) [reply features/upgrade/storage/upgrade.feature:20]: same step is needed in the ckeck case.

File: features/upgrade/storage/upgrade.feature (hunk 5 above)
Comment (qinpingli) [reply features/upgrade/storage/upgrade.feature:63]: same check in the prepare case.

**Pull request closed without merging**